### ISSUE TITLE
HEREDOC highlighting broken

### ISSUE THREAD

**Issue by wyqydsyq** [atom/language-php#57]
I have the following code:

``` php
$this->data['text_message'] .= <<<TRACKER
        <!-- Facebook Conversion Code for Checkout Success -->
<script>(function() {
var _fbq = window._fbq || (window._fbq = []);
if (!_fbq.loaded) {
var fbds = document.createElement('script');
fbds.async = true;
fbds.src ='//connect.facebook.net/en_US/fbds.js';
var s =document.getElementsByTagName('script')[0];
s.parentNode.insertBefore(fbds, s);
_fbq.loaded = true;
}
})();
window._fbq = window._fbq || [];
window._fbq.push(['track', '',{'value':'0.00','currency':'AUD'}]);
</script>
<noscript><img height="1" width="1"alt="" style="display:none" src="https://www.facebook.com/tr?ev=&amp;cd[value]=0.00&amp;cd[currency]=AUD&amp;noscript=1"/></noscript> 
TRACKER;
```

But atom doesn't highlight the contents of the HEREDOC as a string:

![Example](http://i.imgur.com/2F10GqA.png)

The first `<` is correctly given the appropriate classes (`class="string unquoted heredoc php"`) but the class doesn't persist to any of following tokens:
![Example 2](http://i.imgur.com/Jd3Nnyw.png)

**Comment by 50Wliu** [comment]
Refs #10.

**Comment by pyrokinetiq** [comment]
It seems in #10 they're getting the correct behaviour of the heredoc being treated as string until a PHP tag is encountered, my case is only the first character of the heredoc is treated as a string regardless of the heredoc's contents.

**Comment by halleycarleton** [comment]
+1
![HERDOC Syntax Bug]
(https://i.gyazo.com/4dff4b2bb513ca92bae6bc17a4a31f61.png)

*Windows 8.1 / Atom 0.179.0*
*Ubuntu 14.04 / Atom 0.174.0*

**Comment by Talon1024** [comment]
I see that regular expressions are being used to highlight syntax elements like this. Here's a regular expression that matches a PHP Heredoc:
![PHP Heredoc](https://cloud.githubusercontent.com/assets/6550500/7689837/4ec61e2a-fd7d-11e4-9c6e-1d85964a80bc.png)
Here's a regular expression that matches a PHP Nowdoc:
![PHP Nowdoc](https://cloud.githubusercontent.com/assets/6550500/7689901/b67ec2c4-fd7d-11e4-92b7-543408b9b641.png)

**Comment by Krinkle** [comment]
* Atom: v1.0.0
* atom/language-php: v0.26.0

<img width="751" alt="screen shot 2015-07-03 at 02 31 31" src="https://cloud.githubusercontent.com/assets/156867/8491073/b076fd8e-212b-11e5-8f4c-340c34e09493.png">

**Comment by janbaykara** [comment]
Getting the same problem as above, with ' character breaking syntax highlighting. It'd be good if <<<HTML HTML; syntax highlighted packaged code with `language-html`, too.

**Comment by Nicolas-Bouteille** [comment]
any plan for a fix? I find atom really sexy but I use heredocs a lot on my projects. So I am staying on PHPStorm until it is fixed.

**Comment by 50Wliu** [comment]
I may be able to take a look at this soon.  Depends on how quickly I get my existing language backlog cleared up :).

**Comment by Nicolas-Bouteille** [comment]
awesome :)

**Comment by Talon1024** [comment]
This is an issue with [first-mate ignoring nested repositories](https://github.com/atom/first-mate/issues/61).

**Comment by Ener-Getick** [comment]
Any news about this?

**Comment by timbotron** [comment]
Just wanted to chime in that this is an issue for me too. Crossing fingers that someone can get to it soon. Thanks in advance for all the hard work!

**Comment by laviyatziv** [comment]
I'm also seeing this. Any patch to be released?

**Comment by krisj** [comment]
This has been dragging on since issue #10 back in 2014. [https://github.com/atom/language-php/issues/10](https://github.com/atom/language-php/issues/10)

Any joy with making progress on it? Can issues be merged so there is more priority on this?

**Comment by 50Wliu** [comment]
Any help would be welcome.  I'll gladly teach someone the basics if they want to give it a try.

**Comment by Ener-Getick** [comment]
@50Wliu isn't https://github.com/atom/language-php/pull/77 what you want?

**Comment by Skarlso** [comment]
Why has this been closed? It's not fixed, and the issue that it mentioned is also a bug which was closed because of this, but there is no fix?!

**Comment by 50Wliu** [comment]
@Skarlso this issue is clearly still open.

**Comment by Skarlso** [comment]
@50Wliu damn. Argh. Sorry. Messes that one up. :-)

**Comment by 50Wliu** [comment]
Fixed by #184.
